src/relayclient/KnownRelaysManager.ts

```diff
@@ -134,23 +155,30 @@ export default class KnownRelaysManager implements IKnownRelaysManager {
     this.relayFailures = newMap
   }
 
-  async getRelaysSortedForTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayRegisteredEventInfo[]> {
+  async getRelaysSortedForTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayInfoUrl[][]> {
+    const sortedRelays: RelayInfoUrl[][] = []
+    for (let i = 0; i < this.knownRelays.length; i++) {
+      sortedRelays[i] = await this._sortRelaysInternal(gsnTransactionDetails, this.knownRelays[i])
+    }
+    return sortedRelays
+  }
+
+  async _sortRelaysInternal (gsnTransactionDetails: GsnTransactionDetails, activeRelays: RelayInfoUrl[]): Promise<RelayInfoUrl[]> {
     const scores = new Map<string, number>()
-    for (const activeRelay of this.activeRelays) {
-      const score = await this.scoreCalculator(activeRelay, gsnTransactionDetails, this.relayFailures.get(activeRelay.relayUrl) ?? [])
+    for (const activeRelay of activeRelays) {
+      let score = 0
+      if (isInfoFromEvent(activeRelay)) {
+        score = await this.scoreCalculator((activeRelay as RelayRegisteredEventInfo), gsnTransactionDetails, this.relayFailures.get(activeRelay.relayUrl) ?? [])
+      }
       scores.set(activeRelay.relayUrl, score)
```

Comment:
you keep scoring URLs. its wrong. you score managers. these are unique identities (I know, unrelated to preferred relays. its an error entered in previous commit)

Reply:
Well, I will get to fix the keys to be consistent soon